src/dsp/sample_analytics.cpp

```diff
@@ -54,19 +71,34 @@ namespace scxt::dsp {
         };
         }
     */
-        float ms = 0.0f;
-        const float divisor = static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
-        for(size_t i = 0; i < s->getSampleLength(); i++) {
-            for(int chan = 0; chan < s->channels; chan++) {
-                ms += std::powf(*s->GetSamplePtrF32(chan), 2) / divisor;
+    float ms = 0.0f;
+    const float divisor =
+        static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
+    for (size_t i = 0; i < s->getSampleLength(); i++)
+    {
+        for (int chan = 0; chan < s->channels; chan++)
+        {
+            float sample = 0.0;
+            switch (s->bitDepth)
+            {
+            case sample::Sample::BD_I16:
+                sample = static_cast<float>(*s->GetSamplePtrI16(chan)) / std::numeric_limits<int16_t>::max();
```

Comment:
OK so this isn't quite right C++ wise.

GetSamplePtrI16 is the pointer to the data array so doing a * on it is just reading the head element and you are ignoring `i` in the loop. So what you want is something more like

```
static_cast<float>(s->GetSamplePtrI16(chan)[i])
```

and similarly for the F32 below